pkg/api/types/object.go

```diff
@@ -82,76 +82,70 @@ func GetObjectIdentifier(obj Object, singleObjectOperation bool) (string, error)
 	return findIdentifierInStruct(objectStructType, reflect.ValueOf(obj).Elem(), singleObjectOperation)
 }
 
+func extractIdentifierFromField(field reflect.StructField, fieldValue reflect.Value) (*string, error) {
+	if field.Anonymous {
+		embeddedType := field.Type
+		embeddedValue := fieldValue
+
+		for embeddedType.Kind() == reflect.Ptr {
+			embeddedType = embeddedType.Elem()
+			embeddedValue = embeddedValue.Elem()
+		}
+
+		if embeddedType.Kind() == reflect.Struct {
+			id, err := findIdentifierInStruct(embeddedType, embeddedValue, true)
+
+			if errors.Is(err, ErrObjectWithMultipleIdentifier) || errors.Is(err, ErrObjectIdentifierTypeNotSupported) {
+				return nil, fmt.Errorf("type %q: %w", embeddedType, err)
+			} else if !errors.Is(err, ErrObjectWithoutIdentifier) {
```

Comment:
Is it intentional that all other errors are being ignored?

Reply:
fixed that

took this over from before and touching that always made the tests fail, but now found the proper way